src/Glpi/Application/Environment.php

```diff
@@ -0,0 +1,256 @@
+            $value = "";
+        }
+
+        // Try to load the given env, with a fallback to production.
+        return self::tryFrom($value) ?? self::PRODUCTION->value;
+    }
+
+    public static function set(self $environment): void
+    {
+        define('GLPI_ENVIRONMENT_TYPE', $environment->value);
+    }
+
+    public static function validate(): void
+    {
+        // Store valid environments keys
+        $allowed_keys = self::getValues();
+
+        // Validate GLPI_ENVIRONMENT_TYPE if it exists.
+        if (!in_array(GLPI_ENVIRONMENT_TYPE, $allowed_keys)) {
+            throw new UnexpectedValueException(
+                sprintf(
+                    'Invalid GLPI_ENVIRONMENT_TYPE constant value `%s`. Allowed values are: `%s`',
+                    GLPI_ENVIRONMENT_TYPE,
+                    implode('`, `', $allowed_keys)
+                )
+            );
+        }
+    }
+
+    /**
+     * See SystemConfigurator::computeConstants() for all available values that
+     * can be overridden.
+     */
+    public function getConstantsOverride(string $root_dir): array
+    {
+        return match ($this) {
+            default => [],
+            self::TESTING     => [
+                'GLPI_CONFIG_DIR'               => $root_dir . '/tests/config',
+                'GLPI_VAR_DIR'                  => $root_dir . '/tests/files',
+                'GLPI_STRICT_ENV'               => true,
+                'GLPI_SERVERSIDE_URL_ALLOWLIST' => [
+                    // default allowlist entries
+                    '#^http://[^@:]+(:80)?(/.*)?$#',
+                    '#^https://[^@:]+(:443)?(/.*)?$#',
+                    '#^feed://[^@:]+(/.*)?$#',
+
+                    // calendar mockups
+                    '/^file:\/\/.*\.ics$/',
+                ],
+                'PLUGINS_DIRECTORIES'           => [
+                    $root_dir . '/plugins',
+                    $root_dir . '/tests/fixtures/plugins',
+                ],
+            ],
+            self::DEVELOPMENT => [
+                'GLPI_WEBHOOK_ALLOW_RESPONSE_SAVING' => '1'
+            ],
+        };
+    }
+
+    public function getLogLevel(): string
+    {
+        return match ($this) {
+            default           => LogLevel::WARNING,
+            self::TESTING     => LogLevel::DEBUG,
+            self::DEVELOPMENT => LogLevel::DEBUG,
+        };
+    }
+
+    /**
+     * Should debug, info and notice be reported ?
+     * See ErrorHandler::configureErrorReporting().
+     *
+     * Note: this seems weird and should not necessary as we already have
+     * getLogLevel() to define this behavior in the AbstractLogger class.
+     *
+     * TODO: investigate and remove this method and its usage if needed.
+     */
+    public function shouldReportLowLevelErrors(): bool
+    {
+        return match ($this) {
+            default           => false,
+            self::TESTING     => true,
+            self::DEVELOPMENT => true,
+        };
+    }
```

Comment:
You can test it easilly. Keep reporting `E_ALL` in every env and then:
1. Validates that a triggered deprecation is not sent to the logs when env is `production`.
2. Validates that the CI still pass.

Reply:
You must also validate that deprecated messages are not displayed in production when the debug mode is active. Maybe this filtering has been made for this also.